Sources/GraphRepresentation.swift

```diff
@@ -0,0 +1,240 @@
+                renderer.renderScreenActionNode(name: name)
+            }
+        }
+
+        namedScenes.forEach { (src, node) in
+            if let node = node as? MMScreenStateNode {
+                node.edges.values.forEach { edge in
+                    guard let dest = namedScenes[edge.destinationName] else { return }
+                    let directed: Bool
+                    if let dest = dest as? MMScreenStateNode {
+                        directed = !dest.hasBack
+                        renderer.renderEdgeToScreenState(src: src,
+                                                         dest: dest.name,
+                                                         label: edge.predicate?.predicateFormat,
+                                                         isBackable: directed)
+                    } else if let _ = dest as? MMScreenActionNode {
+                        renderer.renderEdgeToScreenAction(
+                            src: src,
+                            dest: dest.name,
+                            label: edge.predicate?.predicateFormat)
+                    } else {
+                        return
+                    }
+
+
+                }
+            } else if let node = node as? MMScreenActionNode {
+                if let destName = node.nextNodeName {
+                    renderer.renderEdgeToScreenState(src: src, dest: destName, label: nil, isBackable: true)
+                }
+            }
+        }
+
+        renderer.end()
+        return renderer.stringValue()
+    }
+}
+
+/// The default implementation
+class DotRepresentation: GraphRepresentation {
+    var lines: [String] = []
+
+    var namedIDs: [String: String] = [:]
+    var idGenerator = 0
+
+    let actionColor = "lightblue"
+    let fontActionColor = "white"
+
+    public var fileExtension: String { return "dot" }
+
+    init() {
+    }
+
+    func begin() {
+        lines = []
+        append("digraph G {",
+               "fontsize=15;",
+               "font=Helvetica;",
+               "labelloc=t;",
+               "label=\"\"", "splines=true", "overlap=false", "rankdir = LR;",
+               "ratio = auto;",
+               "node [ shape = box ]"
+        )
+
+        renderLegend()
+    }
+
+    func stringValue() -> String {
+        return lines.joined(separator: "\n")
+    }
+
+    private func append(_ additional: String...) {
+        lines = lines + additional
+    }
+
+    private func id(for name: String) -> String {
+        if let id = namedIDs[name] {
+            return id
+        }
+
+        let id = "_\(idGenerator)"
+        idGenerator += 1
+        namedIDs[name] = id
+        return id
+    }
+
+
+    func renderScreenStateNode(name: String, isDismissedOnUse dismissOnUse: Bool) {
+        let id = self.id(for: name)
+        var styleCode = "label=\"\(name)\"; "
+        if dismissOnUse {
+            styleCode += "fillcolor=lightgray; color=gray; style=filled"
```

Comment:
nit: might be nice to pull out these colors into global constants, could eventually make it customizable?

Reply:
I've pulled these out into style dictionaries which we merge at render time.